src/users/user-models.js

```diff
@@ -1,39 +1,88 @@
-var mongoose = require('mongoose');
-var Schema = mongoose.Schema;
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const bcrypt = require('bcrypt');
 
-var options = {discriminatorKey: 'type'};
+const options = { discriminatorKey: 'type' };
+const pointOption = {
+  type: Number,
+  required: true,
+  default: 0
+};
 
-var userSchema = new Schema({
-      firstName: String,
-      lastName: String,
-      email: String,
-      role: String,
-    },
-    options);
+const userSchema = new Schema({
+  firstName: String,
+  lastName: String,
+  email: {
+    type: String,
+    unique: true,
+    required: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  role: {
+    type: String,
+    enum: ['Student', 'Admin'],
+    required: true
+  },
+  isApproved: {
+    type: Boolean,
+    required:true,
+    default: false
+  }
+}, options);
+
+userSchema.pre('save', function (next) {
+  const user = this;
+
+  // If the password is not modified, continue saving the user.
+  if (!user.isModified('password')) { return next(); }
+
+  bcrypt.genSalt(Number(process.env.SALT), (err, salt) => {
+    if (err) { return next(err); }
+
+    bcrypt.hash(user.password, salt, (err, hash) => {
+      if (err) { return next(err); }
+
+      user.password = hash;
+      next();
+    });
+  });
+});
+
+userSchema.methods.comparePassword = function (password, next) {
+  bcrypt.compare(password, this.password, (err, isMatch) => {
+    next(err, isMatch);
+  });
+};
 
-var studentSchema = new Schema({
-      approvalStatus: Boolean,
+const studentSchema = new Schema({
       points: {
-        career: Number,
-        community: Number,
-        firstother: Number,
-        firstworkshops: Number,
-        mentor: Number,
-        other: Number,
-        outreach: Number,
-        professor: Number,
-        staff: Number,
-        misc: Number
+        career: pointOption,
+        community: pointOption,
+        firstother: pointOption,
+        firstworkshops: pointOption,
+        mentor: pointOption,
+        other: pointOption,
+        outreach: pointOption,
+        professor: pointOption,
+        staff: pointOption,
+        misc: pointOption
       },
-      studentId: String,
+      studentId: {
+        type: String,
+        unique: true,
+        require: true
+      }
```

Comment:
IIRC, this added as the UID.  If this is still the case, this should be removed, because the UID is created by the database when the document is created.